src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,25 @@
+import { useRef } from "react"
+import { forwardRef, useImperativeHandle } from "react"
+import { createPortal } from "react-dom"
+
+const Modal = forwardRef(function Modal({children}, ref) {
+    const dialog = useRef()
```

Comment:
Its better to rename  this variable   const dialog = useRef(); so it wont make any confusion with the tag.